src/main/java/in/twizmwaz/cardinal/module/modules/wools/WoolObjective.java

```diff
@@ -145,11 +145,7 @@ public void onWoolPickup(InventoryClickEvent event) {
                         this.touched = true;
                         if (touchMessage) {
                             double newProx;
-                            if (location != null) {
-                                newProx = location.distance(place.getVector());
-                            } else {
-                                newProx = player.getLocation().toVector().distance(place.getVector());
-                            }
+                            newProx = player.getLocation().toVector().distance(place.getVector());
```

Comment:
I understand what you're trying to fix with this now, but I think we should use the XML-specified location initially when picking up the wool, so perhaps we should separate out the upcoming `if` statement based on `!oldState`, when the first touch is achieved, and `newProx < proximity`, then the player's distance is less than the current proximity.